### PR TITLE
Allow updating existing checks

### PR DESCRIPTION
**Pull request by nineinchnick**
Allow updating existing checks instead of always creating a new one.

I had to update Github dependencies to access `github.context.job`, which was added a year ago or so.

Without this, there's always a new check run created, with only annotations ([example run](https://github.com/nineinchnick/trino-rest/runs/4385001237?check_suite_focus=true)):
image

With this change, annotations are added to existing jobs, without cluttering the list of jobs ([example run](https://github.com/nineinchnick/trino-rest/actions/runs/1529902876)):
image

### CODE REVIEW

**Comment by ghaiszaher** [comment]
@nineinchnick thank you for your contribution. Could you resolve the conflicts so we can take a look?

**Comment by nineinchnick** [comment]
@ghaiszaher all done!

**Comment by ghaiszaher** [comment]
@nineinchnick thank you. Could you please update from `master` one more time? I fixed the workflow permissions and I need to check if it will run properly now.

Same for the other PR #82

**Comment by nineinchnick** [comment]
Done!

**Review by crypticmind — APPROVED** [review]

**Comment by ghaiszaher** [comment]
Still not working, I can't figure out the correct permissions needed, it could also be related to organization settings. We'll have a look at it later. If you have any ideas let me know.

**Review by ghaiszaher — APPROVED** [review]

**Comment by nineinchnick** [comment]
> Still not working, I can't figure out the correct permissions needed, it could also be related to organization settings. We'll have a look at it later. If you have any ideas let me know.

Workflows running on forks can only have read permissions for the token: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token

This is because they can potentially run untrusted code, without requiring maintainer approval. We're using this action in the Trino project and we had set up a separate workflow to annotate PRs based on test reports saved as artifacts: https://github.com/trinodb/trino/blob/master/.github/workflows/annotate.yml

Notice this is triggered on workflow runs, not PRs, so it executes with a more privileged token.

This probably should be documented here, since it affects anyone that would try to use this action in a public repository.

**Comment by nineinchnick** [comment]
I just found an issue for this: https://github.com/ScaCap/action-surefire-report/issues/50

**Review by renatoabreu11 — APPROVED** [review]

**Comment by ghaiszaher** [comment]
> > Still not working, I can't figure out the correct permissions needed, it could also be related to organization settings. We'll have a look at it later. If you have any ideas let me know.
> 
> Workflows running on forks can only have read permissions for the token: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token
> 
> This is because they can potentially run untrusted code, without requiring maintainer approval. We're using this action in the Trino project and we had set up a separate workflow to annotate PRs based on test reports saved as artifacts: https://github.com/trinodb/trino/blob/master/.github/workflows/annotate.yml
> 
> Notice this is triggered on workflow runs, not PRs, so it executes with a more privileged token.
> 
> This probably should be documented here, since it affects anyone that would try to use this action in a public repository.

@nineinchnick thanks for the insights, let's track this in #50.

**Pull request merged**
